### ISSUE TITLE
Application crashes when calling UpdateManager.RestartApp()

### ISSUE THREAD

**Issue by chikubian** [Squirrel/Squirrel.Windows#431]
Hi @paulcbetts and team, thanks so much for creating Squirrel! It's fantastic!

I am experiencing a crash which *may* be related to the one mentioned by @sickboy in issue #416. 

When my application calls UpdateManager.RestartApp() (after a successful download and apply) the application crashes. Investigating the crash's stack trace, it is caused by a FileNotFoundException searching for (and not finding) Nuget.Squirrel.DLL   I had a similar issue with Mono.Cecil that crashed the app sooner than this, but that problem went away when I manually added a reference to Mono.Cecil using the NuGet package manager.

If you have any clues for what I might be doing wrong, or what I can do to fix it, I'd really appreciate it. I'm not sure if this is relevant, but I'm calling UpdateManager.RestartApp() from within an async method. Does it need to run from an STA thread or something?

Thanks in advance... here is the code in my own UpdateSystem class that is called once the user has decided it is an appropriate moment to download and install the updates that it has detected in another method (this was a necessary usage based on the very short running time of my application, which is itself only a launcher for a bigger product, so we can't rely on background download and update within the applications lifetime...

```csharp
        public async void ApplySoftwareUpdate(bool restartAfterApply, Action<int> progress = null)
        {
            setUpdateStatus(SoftwareUpdateStatus.UpdatesDownloading);
            await _updateManager.DownloadReleases(_updatesAwaitingInstall);

            setUpdateStatus(SoftwareUpdateStatus.UpdatesApplying);
            await _updateManager.ApplyReleases(_updateInfo, progress);

            _updateManager.Dispose();
            UpdateManager.RestartApp();
        }
```
Other information that might be useful: the above code is contained in a supporting DLL which my main application references. It's not directly in the main executable's project. Also, following the crash I've described, the update HAS been applied successfully, so the next time you start my app, it correctly runs the newly updated version.

**Comment by chikubian** [comment]
My apologies, it turns out that it's not UpdateManager.RestartApp() that is encountering the crash. It's the line _updateManager.ApplyReleases(_updateInfo, progress) because even without the call to RestartApp the issue still occurs.

**Comment by ionoy** [comment]
Seems like it's fixed now. If I compile Squirrel from source, then problem goes away. Had the same issue when using NuGet version.

**Comment by paulcbetts** [comment]
Fixed in 1.0.2

**Comment by chikubian** [comment]
Thanks guys. The crash still occurs with the 1.0.2 package from NuGet, however I shall try using the version compiled from source as suggested by @ionoy.

Just a quick question, Mihhail, when you say compile Squirrel from source, do you mean I should create the NUPKG file using the nuspec file in the Squirrel.Windows codebase, or are you suggesting that I manually assemble the outputs from all of the projects (i.e. Setup, Update, Core and Squirrel) into a lib folder and reference them individually? If so, which build configuration should I use, Release or CIBuild?

**Comment by BarryThePenguin** [comment]
@chikubian it should be enough to build the nuget package from source, then  unzip its contents to your projects packages folder

**Comment by chikubian** [comment]
Thanks @BarryThePenguin ... I will give that a try now! Much appreciated!

**Comment by chikubian** [comment]
@BarryThePenguin ... when I try this, and then run the squirrel-generated installer for my application, I get the following popup message:

"This is a dummy update.exe. If you are seeing this, your build did not correctly replace the IDR_UPDATE_ZIP resource. 
Command Line: '--install . '

Have I missed a step? I built this from the latest Squirrel source, using the "Release" build configuration, then ran "nuget pack Squirrel.nuspec" to create the nuget package, and finally extracted the contents of the nuget package to my packages folder as suggested.

**Comment by BarryThePenguin** [comment]
I haven't built Squirrel from source recently, but that's worked for me in the past. I'm guessing you have the sub-modules checked out? `git submodule update --init --recursive`

**Comment by ionoy** [comment]
It seems like this error is still present. I just accidentally copied Nuget.Squirrel.dll to the final build yesterday.

**Comment by kc0isg** [comment]
I don't know what caused it, but when I deleted AppData\Local\SquirrelTemp the error stopped.  I have just been compiling Squirrel and copying NuGet.Squirrel.dll into the same directory as my executable before I package it.

**Comment by chikubian** [comment]
Hi everyone... I really appreciate the assistance. I just managed to get it working with the suggestion from @kc0isg and @ionoy ...

So, building Squirrel (Release build configuration) and copying the NuGet.Squirrel.dll file from its output directory into the same directory as my application's executable before I package it works perfectly. 

I had tried something similar perviously, which didn't work... I tried adding NuGet,Squirrel.dll to my project's references, and this caused Update.exe to complain that it wasn't valid. So it seems that it's important to add the file to your application's output folder *without* referencing it.

FYI, I tried adding an assembly resolver to my application's main AppDomain that mapped references to NuGet.Squirrel.dll or Mono.Cecil.dll into references to Squirrel.dll (or the currently executing assembly), however neither of these helped. I am assuming there are other AppDomains in play here that didn't have access to my AppDomain.AssemblyResolve code. Thanks again for everyone's help!
